### PR TITLE
[6.x] Broadcaster - Improve ShouldBroadcastNow performance

### PR DESCRIPTION
**Pull request by josiasmontag**
### Problem

Currently, `ShouldBroadcastNow` sends the `BroadcastEvent` to the `sync` queue. This causes some performance issues: 
* It [serializes](https://github.com/laravel/framework/blob/5b1b3675748649da19c9b6308d1ade25f41eabd5/src/Illuminate/Queue/Queue.php#L139) and directly afterwards [unserializes](https://github.com/laravel/framework/blob/a6edb86ebbeb21798ac7566c9b6fd3c94929c3c6/src/Illuminate/Events/CallQueuedListener.php#L142) the `BroadcastEvent`
* In combination with the `SerializesModels` Trait it creates completely unnecessary database queries

### Solution

We could just directly `dispatchNow()` the `BroadcastEvent`. There should be no reason at all to send it to the the `sync` queue.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Broadcast manager wasn't being tested so I added some.

File: src/Illuminate/Broadcasting/BroadcastManager.php [hunk 1]
```diff
@@ -124,7 +123,7 @@ public function queue($event)
             $queue = $event->queue;
         }
 
-        $this->app->make('queue')->connection($connection)->pushOn(
+        $this->app->make('queue')->connection($event->connection ?? null)->pushOn(
```
Comment (rimace) [inline src/Illuminate/Broadcasting/BroadcastManager.php]: Can I ask why `$event->connection ?? null` is used?
I may misunderstand the `??` operator, but as I understand it it checks if the connection is null and if it is, don't take it but the second "argument", which is a hard written `null` here.

Comment (GrahamCampbell) [reply src/Illuminate/Broadcasting/BroadcastManager.php]: In the event that $event doesn't have a connection property, the `??` suppresses the error:

image

Comment (rimace) [reply src/Illuminate/Broadcasting/BroadcastManager.php]: Thanks, didn't know it also checks `$event` itself.

**Comment by mfn** [comment]
Small PSA:

I'm in the process of upgrading and this broke my tests from Laravel 5.8. Previously broadcast events where collected by the QueueFake but now they are disaptched.

In Laravel 5.8 `Queue::fake()` works because `\Illuminate\Broadcasting\BroadcastManager::queue` in fact uses the queue:
```php
        $this->app->make('queue')->connection($connection)->pushOn(
            $queue, new BroadcastEvent(clone $event)
        );

```
In the above, `make('queue')` would return the `\Illuminate\Support\Testing\Fakes\QueueFake` and thus _collect_ the broadcast event and I can assert on that, instead the event gets fired now.

The solution is to now use the `Bus::fake()` on such tests and use the `assertDispatched` thereon.

Note: the same break would be true when upgrading from any Laravel version <  6.7

**Comment by driesvints** [comment]
@mfn good catch. But I think it's best that we leave it as is here.

**Comment by mfn** [comment]
@driesvints yes! Not suggesting a change, just wanted to make something google-worthy for others 😄

**Pull request merged**